packages/@azure/servicebus/data-plane/lib/session/messageSession.ts

```diff
@@ -590,6 +593,10 @@ export class MessageSession extends LinkEntity {
             }
           }
           return;
+        } finally {
+          if (this._receiver) {
+            this._receiver!.addCredit(1);
```

Comment:
Since we are not yet clear on how credits work, should we keep the credit related changes out of this PR?

If we do so, we can first concentrate on the batching scenario, get this PR merged and then have another PR for streaming.

Reply:
All of the tests are passing with these changes, and credit handling in general may need more experimenting to better understand as documentation is limited. I think it may be fine to have these in and work on the credit handling as a separate issue?

Reply:
I still wouldnt want to change anything in an area unless we understand exactly how it works.